### PR TITLE
git-hash-object: add page

### PR DESCRIPTION
**Pull request by nath1as**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

### CODE REVIEW

**Comment by nath1as** [comment]
PR for the https://github.com/tldr-pages/tldr/issues/3953 ticket

File: pages/common/git-hash-object.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# git hash-object
+
+> Computes the unique hash key of content and optionally creates an object with specified type.
+> More information: <https://git-scm.com/docs/git-hash-object>.
+
+- Compute the ID that would be used to store it in git database:
```
Comment (kbdharun) [inline pages/common/git-hash-object.md:6]:
```suggestion
- Compute the ID used to store it in the git database:
```
I think this would be better.

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for this page.

File: pages/common/git-hash-object.md (hunk 1 above)
Comment (nath1as) [reply pages/common/git-hash-object.md:6]: I wanted to express that it isn't stored in the git database, but returns the value that would be stored if we wanted it (by providing -w option).

File: pages/common/git-hash-object.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+`git hash-object {{/path/to/file}}`
+
+- Compute the ID and store it in git database:
+
+`git hash-object -w {{/path/to/file}}`
+
+- Compute the ID and specify the type of object:
+
+`git hash-object -t {blob | tree | commit | tag} {{/path/to/file}}`
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:16]:
```suggestion
`git hash-object -t {{blob|commit|tag|tree}} {{path/to/file}}`
```

File: pages/common/git-hash-object.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# git hash-object
+
+> Computes the unique hash key of content and optionally creates an object with specified type.
+> More information: <https://git-scm.com/docs/git-hash-object>.
+
+- Compute the ID that would be used to store it in git database:
+
+`git hash-object {{/path/to/file}}`
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:8]:
```suggestion
`git hash-object {{path/to/file}}`
```

File: pages/common/git-hash-object.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+- Compute the ID and store it in git database:
+
+`git hash-object -w {{/path/to/file}}`
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:12]:
```suggestion
`git hash-object -w {{path/to/file}}`
```

File: pages/common/git-hash-object.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+- Compute the ID and specify the type of object:
+
+`git hash-object -t {blob | tree | commit | tag} {{/path/to/file}}`
+
+- Compute the ID from user input instead of a file:
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:18]:
```suggestion
- Compute the ID from stdin:
```

File: pages/common/git-hash-object.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+`git hash-object --stdin`
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:20]: This is better
```suggestion
`cat {{path/to/file}} | git hash-object --stdin`
```

File: pages/common/git-hash-object.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# git hash-object
+
+> Computes the unique hash key of content and optionally creates an object with specified type.
+> More information: <https://git-scm.com/docs/git-hash-object>.
+
+- Compute the ID that would be used to store it in git database:
+
+`git hash-object {{/path/to/file}}`
+
+- Compute the ID and store it in git database:
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:10]:
```suggestion
- Compute the object ID and store it in the Git database:
```

File: pages/common/git-hash-object.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+`git hash-object -w {{/path/to/file}}`
+
+- Compute the ID and specify the type of object:
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:14]:
```suggestion
- Compute the object ID specifying the object type:
```

File: pages/common/git-hash-object.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+- Compute the object ID and store it in the Git database:
+
+`git hash-object -w {{path/to/file}}`
+
+- Compute the object ID specifying the object type:
+
+`git hash-object -t {{blob|commit|tag|tree}} {{path/to/file}}`
+
+- Compute the ID from stdin:
```
Comment (navarroaxel) [inline pages/common/git-hash-object.md:18]:
```suggestion
- Compute the object ID from stdin:
```

File: pages/common/git-hash-object.md (hunk 1 above)
Comment (navarroaxel) [reply pages/common/git-hash-object.md:6]: I think a simple English is better here
```suggestion
- Compute the object ID without store it:
```

File: pages/common/git-hash-object.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# git hash-object
+
+> Computes the unique hash key of content and optionally creates an object with specified type.
+> More information: <https://git-scm.com/docs/git-hash-object>.
+
+- Compute the object ID without store it:
```
Comment (nath1as) [inline pages/common/git-hash-object.md:6]: storing

**Review by navarroaxel — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Pull request merged**